org.eclipse.scanning.event.queues/src/org/eclipse/scanning/event/queues/QueueService.java

```diff
@@ -81,24 +83,42 @@
 	
 	/**
 	 * No argument constructor for OSGi
+	 * @throws URISyntaxException 
 	 */
 	public QueueService() {
-		
+		this(getQueueRootFromProperty(), CommandConstants.getScanningBrokerUri());
+	}
+	
+	private static final String getQueueRootFromProperty() {
+		String root = System.getProperty("org.eclipse.scanning.event.queues.queue.root");
+		if (root==null) root = System.getProperty("GDA/gda.event.queues.queue.root");
+		return root;
 	}
 	
+	public QueueService(String queueRoot, URI uri) {
+		this.queueRoot = queueRoot;
+		this.uri = uri;
+	}
 	/**
 	 * Constructor for tests
 	 */
-	public QueueService(String queueRoot, URI uri) {
-		QueueService.queueRoot = queueRoot;
-		QueueService.uri = uri;
+	public QueueService(String queueRoot, String uri) {
+		this.queueRoot = queueRoot;
+		this.stringUri = uri;
 	}
 
-	@PostConstruct
 	@Override
 	public void init() throws EventException {
 		//Check configuration is present
 		if (queueRoot == null) throw new IllegalStateException("Queue root has not been specified");
+		
+		if (uri==null && stringUri!=null) {
```

Comment:
Rather than have the URI be set here, if you call setUri(_string_) in the constructor, all this will get run for you and you can remove the line after the next comment:
`uriString = uri.toString();`